test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -177,6 +257,76 @@ TEST_F(PickFirstTest, GoesIdleWhenConnectionFailsThenCanReconnect) {
   }
 }
 
+TEST_F(PickFirstTest, WithShuffle) {
+  testing::ScopedExperimentalEnvVar env_var(
+      "GRPC_EXPERIMENTAL_PICKFIRST_LB_CONFIG");
+  // 6 addresses have 6! = 720 permutations or 0.1% chance that the shuffle
+  // returns initial sequence
+  constexpr std::array<absl::string_view, 6> kAddresses = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  absl::Status status = ApplyUpdate(
+      BuildUpdate(kAddresses, MakePickFirstConfig(true)), lb_policy_.get());
+  EXPECT_TRUE(status.ok()) << status;
+  std::vector<absl::string_view> prev_attempt_connect_order;
+  GetOrderAddressesArePicked(kAddresses, &prev_attempt_connect_order);
+  size_t first_shuffle = 0;
+  for (size_t i = 0; i < kAddresses.size(); ++i) {
```

Comment:
I don't think we need to iterate over the individual elements; instead, we can just compare the vectors as a whole.  We also don't need to handle the initial attempt differently from the subsequent attempts.  I think we can just do this:

```
std::vector<absl::string_view> prev_order(kAddresses.begin(), kAddresses.end());
for (size_t attempt = 0; attempt < 3; ++attempt) {
  std::vector<absl::string_view> new_order;
  GetOrderAddressesArePicked(kAddresses, &new_order);
  EXPECT_NE(new_order, prev_order);
  prev_order = std::move(new_order);
}
```

Same thing for the next test, except that it will use `EXPECT_EQ()` instead of `EXPECT_NE()`.